web/rockblock.go

```diff
@@ -0,0 +1,147 @@
+package web
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/SmartParksOrg/smartparks-connect-web/utils"
+)
+
+type RockBLOCKRequest struct {
+	RequestType string `json:"request_type"`
+	Username    string `json:"username"`
+	Password    string `json:"password"`
+	Port        uint8  `json:"port"`
+	IMEI        string `json:"imei"`
+	Confirmed   bool   `json:"confirmed"`
+
+	ID      string      `json:"id"`
+	Length  uint8       `json:"content_length"`
+	Type    string      `json:"content_type"`
+	Content interface{} `json:"content"`
+}
+
+type RockBLOCKLoginRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+func (h *Handler) handleRockBLOCKLogin(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+	decoder := json.NewDecoder(r.Body)
+	var request RockBLOCKLoginRequest
+	err := decoder.Decode(&request)
+	if err != nil {
+		http.Error(w, "Invalid request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Create a new HTTP POST request with the RockBLOCK API endpoint URL
+	url := fmt.Sprintf("https://rockblock.rock7.com/rockblock/MT?username=%s&password=%s&data=0",
+		url.QueryEscape(request.Username),
+		url.QueryEscape(request.Password),
+	)
+	req, _ := http.NewRequest("POST", url, nil)
+	req.Header.Add("accept", "text/plain")
+
+	// Send the request and parse the response
+	res, err := h.httpClient.Do(req)
+	if err != nil {
+		Resp(w, nil, err)
+		return
+	}
+	defer res.Body.Close()
+	body, _ := io.ReadAll(res.Body)
+
+	// Check the response for errors
+	if strings.HasPrefix(string(body), "FAILED") {
+		// Response is an error, get the code from the response message
+		parts := strings.SplitN(string(body), ",", 3)
+		code, _ := strconv.Atoi(parts[1])
+		if code == 10 {
+			Resp(w, nil, errors.New(parts[2]))
+			return
+		}
+		// Other error codes are ignored at this stage, since we are not actually sending data to any device, we just
+		// want to check if credentials are correct
+	}
+
+	Resp(w, nil, nil)
+}
+
+func (h *Handler) handleRockBLOCKQueue(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	if r.Method != "POST" {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Decode JSON data
+	decoder := json.NewDecoder(r.Body)
+	var request RockBLOCKRequest
+	err := decoder.Decode(&request)
+	if err != nil {
+		http.Error(w, "Invalid request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	if request.Port == 0 || request.ID == "" || request.Type == "" || request.Length == 0 || request.Content == "" {
+		http.Error(w, "missing one or more of required parameters: port, content, content_type, id, type", http.StatusBadRequest)
```

Comment:
Currently, it hits this error when trying to send a cmd_send_status.
It was difficult to debug as the front end didn't display what the error was , just code 400 and no logs from the backend as well
Can we use unify all responses to use the Resp function and log everything that is happening inside and also the front end to return the actual errors.

Also I don't think we need these. Instead  Rest seems to handle when err is nil and not nil
 
```
func Succ(writer http.ResponseWriter, result interface{}) {
	Resp(writer, result, nil)
}

func Err(write http.ResponseWriter, err error) {
	Resp(write, nil, err)
}
```

Reply:
Those functions were there before already so I didn't touch them.

Regarding usage of Resp, I'll use that, but it's against the standard, since the actual response code returned is always 200, which is not really good practice.

Reply:
This is an overlook at my side. Please change it so that it returns the appropriate HTTP codes.